modxcms/revolution
# Improve the processor permission response error message

### What does it do?
- Add a lexicon entry for the processor permission message.
- Output the permission and the processor action in the message (Moving the permission class property to the modProcessor class was needed).
- Show the new error message in the MODx.combo.ComboBox loadexception

### Why is it needed?
It is not easy to detect, which permission is needed for a combobox processor permission error.

Before:

![Bildschirmfoto 2021-02-20 um 21 13 52](https://user-images.githubusercontent.com/148371/108609041-890fc700-73cb-11eb-854e-443859b08ee7.png)

After:

![Bildschirmfoto 2021-02-20 um 21 14 42](https://user-images.githubusercontent.com/148371/108609048-8f05a800-73cb-11eb-934d-dd8d5b7d7303.png)

### How to test
Add the following code to i.e. core/model/modx/processors/workspace/namespace/getlist.class.php and open the MODX system settings.

```
    public function checkPermissions() {
        return false;
    }
```

In a grid, the message is shown in the empty grid content.

### Related issue(s)/PR(s)
#13243, #12568, #15344
